core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1492,7 +1495,7 @@ public ConnectionCheckJob(UpdateSite site) {
         public void run() {
             connectionStates.put(ConnectionStatus.INTERNET, ConnectionStatus.UNCHECKED);
             connectionStates.put(ConnectionStatus.UPDATE_SITE, ConnectionStatus.UNCHECKED);
-            if (ID_UPLOAD.equals(site.getId())) {
+            if (site == null || ID_UPLOAD.equals(site.getId())) {
```

Comment:
Does it deserve a log warning for the new check? Maybe also a separate PR?

Reply:
I do not think so. The NPE was observed only in a CloudBees update site configuration and is related to the new job type, which is not specific to a site.